src/components/views/read/cvForm/PersonalInfo.tsx

```diff
@@ -0,0 +1,350 @@
+import dayjs, { Dayjs } from 'dayjs';
+
+import { FC, RefObject, useRef, useState } from 'react';
+import PhoneInput from 'react-phone-input-2';
+import 'react-phone-input-2/lib/style.css';
+
+import NavigateBeforeIcon from '@mui/icons-material/NavigateBefore';
+import NavigateNextIcon from '@mui/icons-material/NavigateNext';
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import { Button, MenuItem, Select, TextField } from '@mui/material';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+
+interface InnerObject {
+  [key: string]: string;
+}
+
+interface ValuesObject {
+  [key: string]: InnerObject;
+}
+
+interface HandleModifyFunction {
+  (category: string, modifiedValues: any): void;
+}
+interface Props {
+  nextPage: () => void;
+  prevPage: () => void;
+  nextStep: () => void;
+  prevStep: () => void;
+  values: ValuesObject;
+  handleValues: HandleModifyFunction;
+}
+
+const PersonalInfo: FC<Props> = ({
+  nextPage,
+  prevPage,
+  nextStep,
+  prevStep,
+  values,
+  handleValues,
+}) => {
+  const handlePrev = (): void => {
+    prevPage();
+    prevStep();
+  };
+  const [birthDate, setBirthDate] = useState<Dayjs | null>(dayjs('2022-04-17'));
+  const [phoneNumber, setPhoneNumber] = useState('');
+  const inputRef: RefObject<HTMLInputElement> = useRef(null);
+  const handleClick = (): void => {
+    // 👇️ open file input box on click of another element
+    inputRef.current?.click();
+  };
+  const [inputValues, setInputValues] = useState<{ [key: string]: string }[]>(
+    [],
+  );
+  const handleChange = (field: string, value: string): void => {
+    setInputValues((prevInputValues) => {
+      const updatedInputValues = [...prevInputValues];
+      const fieldValue = { [field]: value };
+      const existingFieldIndex = updatedInputValues.findIndex(
+        (inputValue) => Object.keys(inputValue)[0] === field,
+      );
+
+      if (existingFieldIndex !== -1) {
+        // Field already exists, update its value
+        updatedInputValues[existingFieldIndex] = fieldValue;
+      } else {
+        // Field doesn't exist, add it to the input values
+        updatedInputValues.push(fieldValue);
+      }
+
+      return updatedInputValues;
+    });
```

Comment:
Why are you pushing single-keyed objects into your `inputValues` array ?? 
Why don't you use a simple object and update the keys ? 
Currently: 
```ts
[
  {firstname: "<value for first name>"},
  {lastname: "<value for last name>"},
  {gender: "<value for gender>"},
  ...,
]
```
What you should do:
```ts
{ 
  firstname: "<value for first name>",
  lastname: "<value for last name>",
  gender: "<value for gender>",
  ...,
}
```